test/expect-tests/fiber/fiber_tests.ml

```diff
@@ -847,6 +847,60 @@ let%expect_test "raise exception" =
        (fun () -> Pool.stop pool));
   [%expect {| Caught Exit |}]
 
+let%expect_test "double run a pool" =
+  (* Calling [Pool.run] on the same pool shouldn't be allowed *)
```

Comment:
Could you explain why it shouldn't be allowed? (I don't know much about the pools, sorry.)

Reply:
Currently, `Pool.run` means to run all the tasks inside the pool until a `stop` has been given.

So if we run two `Pool.run`, they will compete for the same queue of tasks and it will be unclear which `run` will get which task. This is really important when tasks raise exceptions as we want to know exactly where uncaught task exceptions should be caught (around `run`).

Reply:
Great! Please copy this text as a comment.